File: Simplified/NYPLFacetBarView.m
#import "NYPLFacetView.h"
#import "SimplyE-Swift.h"
#import <PureLayout/PureLayout.h>

#import "NYPLFacetBarView.h"

@interface NYPLFacetBarView ()

@property (nonatomic) NYPLEntryPointView *entryPointView;
@property (nonatomic) NYPLFacetView *facetView;

@end

@implementation NYPLFacetBarView

- (instancetype)initWithOrigin:(CGPoint const)origin width:(CGFloat const)width
{
  CGFloat const borderHeight = 1.0 / [UIScreen mainScreen].scale;
  CGFloat const toolbarHeight = 40;
  
  self = [super initWithFrame:CGRectMake(origin.x, origin.y, width, borderHeight + toolbarHeight)];
  if(!self) return nil;

  UIVisualEffect *blur = [UIBlurEffect effectWithStyle:UIBlurEffectStyleExtraLight];
  UIVisualEffectView *bgBlur = [[UIVisualEffectView alloc] initWithEffect:blur];

  [self addSubview:bgBlur];
  [bgBlur autoPinEdgesToSuperviewEdges];

  self.entryPointView = [[NYPLEntryPointView alloc] init];
  self.facetView = [[NYPLFacetView alloc] init];
  self.entryPointView.hidden = YES;
  self.facetView.hidden = YES;

  UIView *bottomBorderView = [[UIView alloc] init];
  bottomBorderView.backgroundColor = [[UIColor lightGrayColor] colorWithAlphaComponent:0.9];
  UIView *topBorderView = [[UIView alloc] init];
  topBorderView.backgroundColor = [[UIColor lightGrayColor] colorWithAlphaComponent:0.9];

  [self addSubview:self.facetView];
  [self addSubview:self.entryPointView];
  [self.entryPointView autoPinEdgesToSuperviewEdgesWithInsets:UIEdgeInsetsZero excludingEdge:ALEdgeBottom];
  [self.facetView autoPinEdgesToSuperviewEdgesWithInsets:UIEdgeInsetsZero excludingEdge:ALEdgeTop];
  [self.entryPointView autoPinEdge:ALEdgeBottom toEdge:ALEdgeTop ofView:self.facetView];

  [self.facetView addSubview:bottomBorderView];
  [bottomBorderView autoPinEdgesToSuperviewEdgesWithInsets:UIEdgeInsetsZero excludingEdge:ALEdgeTop];
  [bottomBorderView autoSetDimension:ALDimensionHeight toSize:borderHeight];
  [self.facetView addSubview:topBorderView];
  [topBorderView autoPinEdgesToSuperviewEdgesWithInsets:UIEdgeInsetsZero excludingEdge:ALEdgeBottom];
  [topBorderView autoSetDimension:ALDimensionHeight toSize:borderHeight];

  return self;
}

@end

File: Simplified/Views/NYPLFacetBarView.swift
import Foundation

@objcMembers class NYPLFacetBarView : UIView {
  var entryPointView: NYPLEntryPointView;
  var facetView: NYPLFacetView;
  
  @available(*, unavailable)
  private override init(frame: CGRect) {
    self.entryPointView = NYPLEntryPointView()
    self.facetView = NYPLFacetView()
    
    super.init(frame: frame)
  }
  
  @available(*, unavailable)
  required init?(coder: NSCoder) {
    fatalError("init(coder:) has not been implemented")
  }
  
  init(origin: CGPoint, width: CGFloat) {
    self.entryPointView = NYPLEntryPointView()
    self.facetView = NYPLFacetView()
    
    let borderHeight = 1.0 / UIScreen.main.scale;
    let toolbarHeight = CGFloat(40);
    
    super.init(frame: CGRect(x: origin.x, y: origin.y, width: width, height: borderHeight + toolbarHeight))
    
    var blur: UIVisualEffect;
    if #available(iOS 13, *) {
      blur = UIBlurEffect(style: .prominent);
    } else {
      blur = UIBlurEffect(style: .extraLight);
    }
    let bgBlur = UIVisualEffectView(effect: blur)

    addSubview(bgBlur)
    bgBlur.autoPinEdgesToSuperviewEdges()

    self.entryPointView.isHidden = true;
    self.facetView.isHidden = true;

    let bottomBorderView = UIView()
    bottomBorderView.backgroundColor = UIColor.lightGray.withAlphaComponent(0.9)
    let topBorderView = UIView()
    topBorderView.backgroundColor = UIColor.lightGray.withAlphaComponent(0.9)

    addSubview(self.facetView)
    addSubview(self.entryPointView)
    self.entryPointView.autoPinEdgesToSuperviewEdges(with: UIEdgeInsets.zero, excludingEdge: .bottom)
    self.facetView.autoPinEdgesToSuperviewEdges(with: UIEdgeInsets.zero, excludingEdge: .top)
    self.entryPointView.autoPinEdge(.bottom, to: .top, of: self.facetView)

    self.facetView.addSubview(bottomBorderView)
    bottomBorderView.autoPinEdgesToSuperviewEdges(with: UIEdgeInsets.zero, excludingEdge: .top)
    bottomBorderView.autoSetDimension(.height, toSize: borderHeight)
    self.facetView.addSubview(topBorderView)
    topBorderView.autoPinEdgesToSuperviewEdges(with: UIEdgeInsets.zero, excludingEdge: .bottom)
    topBorderView.autoSetDimension(.height, toSize:borderHeight)
  }
}
